### PR TITLE
CI and Example Upgrade to Dojo 1.6.1 and Scarb 2.10.1

### PR DESCRIPTION
**Pull request by okhaimie-dev**
# Pull Request Overview

## 📝 Summary
This PR upgrades the ci and also project dojo and scarb dependency to `1.6.1` and `2.10.1` respectively.

### Related Issues
- Closes https://github.com/AkatsukiLabs/DojoByExample/issues/176

### Type of Change
Mark with an `x` all the checkboxes that apply (like `[x]`).
- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

### What's Changed
- I decided to install dojo directly from binary to reduce the need of unnecessary dependency for ci. 
- I also separated the `fmt --check` from the rest of the job for faster feedback, developers will not have to wait till the end of the first job just to know that their format was wrong. 

### Implementation Details
Right now, you don't need to install asdf before installing dojo, all you need to do is change the environmental variable of Dojo and Scarb and removes the worry of having to also maintain asdf.

### Technical Notes
Include any technical details that reviewers should be aware of.

## 🔧 Tests Results
Describe the tests you performed to verify your changes.

### Test Coverage
- [x] Unit Tests
- [x] Integration Tests
- [x] Manual Testing

### Evidence
CleanShot 2025-07-26 at 18 05 36@2x

### Testing Notes
Include any special testing considerations or edge cases checked.

## 🔜 Next Steps
Indicate actions or improvements to be taken after this PR, if applicable.

### CODE REVIEW

**Comment by okhaimie-dev** [comment]
@coxmars I modified somethings.

Which I believe makes the CI more maintainable and efficient and also used the latest compatible version of dojo `1.6.1` instead of the alpha version

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,57 +2,53 @@ name: Cairo-Dojo CI
...
-    branches: [ main, master ]
+    branches: [main]
   pull_request:
-    branches: [ main, master ]
+    branches: [main]
+
+env:
+  DOJO_VERSION: 1.6.1
+  SCARB_VERSION: 2.10.1
+  WORKING_DIRECTORY: ./backend/dojo_examples/combat_game
```
Comment (okhaimie-dev) [inline .github/workflows/ci.yml:12]: In the future y'all just need to change the env variable to upgrade the dojo and scarb version

**Comment by jimenezz22** [comment]
@coxmars check this for merge
